File: Sorting-Algorithms/Insertion_Sort.py
# standard library 
import sys

def ReadFile(filename):
	with open(filename) as file:
		data = file.read()
		data = data.split(' ')
		Input = data
		Input = list(map(int, Input))
		file.close()
	return Input

def sort(Input, size):
	for x in range( size):
		for y in range(size):
			if Input[x] < Input[y]:
				temp = Input[y]
				Input[y] = Input[x]
				Input[x] = temp
	print (Input)
def main():
	file = input("Please enter file to sort: ")
	Input = ReadFile(file)
	
	#Input = [1,5,3,2,9, 6, 6, 0]

	size = len(Input)
	sort(Input, size)


main()

File: Sorting-Algorithms/Insertion_Sort.cpp
#include <iostream>

int main(){
	int Input[] = {1, 2, 6, 9, 8, 2, 5};
	int size = (sizeof(Input)/sizeof(int));
	int temp = 0;
	for (int i = 0; i < size; i++)
	{

		int key = Input[i];
			temp = Input[i];
		for (int j = 0; j < (size); j++)
		{
			if (key <= Input[j])
				{
				temp = Input[j];
				Input[j] = Input[i];
				Input[i] = temp;
				}

		}
	}

	for (int i = 0; i < size; i++)
	{
		std::cout << Input[i];
	}

	return 0;
}
